dapp_runner/descriptor/dapp.py

```diff
@@ -110,7 +119,37 @@ def __implicit_vpn(self):
                     vm.VM_CAPS_VPN
                 ]
 
+    def _resolve_dependencies(self):
+        """Resolve instantiation priorities."""
+
+        # initialize the dependency graph and add a root node
+        self._dependency_graph: networkx.DiGraph = networkx.DiGraph()
+        self._dependency_graph.add_node(DEPENDENCY_ROOT)
+
+        # for now, we only care about the order of services,
+        # later we can enhance the dependency graph to
+        # take all the other entities into consideration
+
+        for name, service in self.nodes.items():
+            if service.depends_on:
+                self._dependency_graph.add_edge(name, service.depends_on)
+            else:
+                self._dependency_graph.add_edge(DEPENDENCY_ROOT, name)
+
+        if not networkx.is_directed_acyclic_graph(self._dependency_graph):
+            raise DescriptorError(
+                "Service definition contains a circular `depends_on`."
+            )
+
+    def nodes_prioritized(self) -> Generator[Tuple[str, ServiceDescriptor], None, None]:
+        """Get a dict-items-like generator of services, ordered by dependencies."""
+
+        for name in reversed(list(networkx.topological_sort(self._dependency_graph))):
```

Comment:
Is `list` necessary here?

Reply:
yes, `reversed` doesn't accept a generator...